test/integration/scheduler_perf/scheduler_test.go

```diff
@@ -253,3 +237,121 @@ func schedulePods(config *testConfig) int32 {
+			NodeAffinity: &v1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+					NodeSelectorTerms: []v1.NodeSelectorTerm{
+						{
+							MatchExpressions: []v1.NodeSelectorRequirement{
+								{
+									Key:      nodeAffinityKey,
+									Operator: nodeAffinity.Operator,
+									Values:   []string{fmt.Sprintf("%v", i)},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		podList = append(podList, pod)
+	}
+	return podList
+}
+
+// readInConfiguration reads the input parameters for every predicate and priority that are needed for test run.
+// TODO: As of now, returning configs hardcoded, need to read from yaml or some other file.
+func readInConfiguration() *schedulerPerfConfig {
+	return &schedulerPerfConfig{
+		NodeAffinity: nodeAffinity{
+			Enabled:         false,
+			numGroups:       10,
+			nodeAffinityKey: "kubernetes.io/sched-perf-node-affinity",
+			Operator:        v1.NodeSelectorOpIn,
+		},
+
+		InterpodAffinity: interpodAffinity{
+			Enabled:     false,
+			Operator:    metav1.LabelSelectorOpIn,
+			affinityKey: "security",
+			Labels:      map[string]string{"security": "S1"},
+			TopologyKey: "region",
+		},
+	}
+}
+
+// mutate is the wrapper function for modifying both pods and nodes.
+func (inputConfig *schedulerPerfConfig) mutate(config *testConfig) {
+	nodeAffinity := inputConfig.NodeAffinity
+	podCreatorConfig := testutils.NewTestPodCreatorConfig()
+	var nodeStrategies []testutils.CountToStrategy
+	var pod *v1.Pod
+	var podList []*v1.Pod
+	if nodeAffinity.Enabled {
+		// Mutate Node
+		nodeStrategies = nodeAffinity.mutateNode(config.numNodes)
+		// Mutate Pod TODO: Make this to return to podSpec.
+		podList = nodeAffinity.mutatePod(config.numPods, pod)
+		numGroups := nodeAffinity.numGroups
+		for _, pod := range podList {
+			podCreatorConfig.AddStrategy("sched-perf-node-affinity", config.numPods/numGroups,
+				testutils.NewCustomCreatePodStrategy(pod),
+			)
+		}
+		config.nodePreparer = framework.NewIntegrationTestNodePreparer(
+			config.schedulerSupportFunctions.GetClient(),
+			nodeStrategies, "scheduler-perf-")
+		config.podCreator = testutils.NewTestPodCreator(config.schedulerSupportFunctions.GetClient(), podCreatorConfig)
+	} else {
+		// Default configuration.
+		nodePreparer := framework.NewIntegrationTestNodePreparer(
+		config.schedulerSupportFunctions.GetClient(),
+		[]testutils.CountToStrategy{{Count: config.numNodes, Strategy: &testutils.TrivialNodePrepareStrategy{}}},
+		"scheduler-perf-",
+		)
+
+		podConfig := testutils.NewTestPodCreatorConfig()
+		podConfig.AddStrategy("sched-test", config.numPods, testutils.NewSimpleWithControllerCreatePodStrategy("rc1"))
+		podCreator := testutils.NewTestPodCreator(config.schedulerSupportFunctions.GetClient(), podConfig)
+		config.nodePreparer = nodePreparer
+		config.podCreator = podCreator
+	}
+	return
+}
+
+
+
+// For initializing the variables.
+func parseInput(config *testConfig) {
```

Comment:
Since parse isn't doing any parsing, why not just delete this for now, and move these two lines of code into the individual tests?